victorLi/assignments/treesBFS/lc102/lc102.js

```diff
@@ -0,0 +1,122 @@
+// Problem Statement #
+
+// Given a binary tree, populate an array to represent its level-by-level traversal. You should populate the values of all nodes of each level from left to right in separate sub-arrays.
+
+
+/*
+               12
+           7        1
+        9         10    5
+
+Queue array keep track of the current level node to be process
+Level Array to store node by lvl respectively
+QueueLength variable
+
+Outerloop until queue.length >0
+  {
+     set queueLength variable
+     reset LevelArray
+          InnerLoop queueLength > 0
+          {
+            Shift queue and store the current node
+            check current node left and right node if found push into queue
+            push current node in levelArray
+            
+          }
+        push levelArray into result array
+
+        }
+
+        return result array
+
+*/
+
+
+
+
+
+class TreeNode {
+
+  constructor(value) {
+    this.value = value;
+    this.left = null;
+    this.right = null;
+  }
+};
+
+
+const traverse = function(root) {
+    // TODO: Write your code here
+  result = [];
+  let queue = [root]
+ 
+
+  while( queue.length)
+  {
+    let length = queue.length;
+    let levelArray = []
+     for( i = 0; i < length; i++ )
+     {
+       let node = queue.shift()
+       if( node.left) queue.push(node.left)
+       if( node.right) queue.push(node.right)
+       levelArray.push(node.value)
+     }
+    result.push(levelArray)
+
+  }
+
+  return result;
+};
+
+
+
+var root = new TreeNode(12);
+root.left = new TreeNode(7);
+root.right = new TreeNode(1);
+root.left.left = new TreeNode(9);
+root.right.left = new TreeNode(10);
+root.right.right = new TreeNode(5);
+console.log(`Level order traversal: ${traverse(root)}`);
+
```

Comment:
Could be helpful to write additional test cases - especially for any edge cases.